tests/MongoDB.Bson.Tests/ObjectModel/Decimal128Tests.cs

```diff
@@ -87,6 +90,19 @@ public void ToDecimal_should_return_expected_result(string valueString, string e
             result.Should().Be(expectedResult);
         }
 
+        [Theory]
+        [InlineData(0xE000_2000_0000_0000UL, 0)]
+        public void ToDecimal_should_return_expected_result_for_second_form(ulong highBits, ulong expectedResult)
+        {
+            // todo: does this logic for processing `second form` make sense?
```

Comment:
I'm not sure how this logic was expected to work?
NOTE: `Second form` is 128-bit value which is constructed based on this template:
>the sign bit (1 bit) with the combination bits (17 bits) with the significand continuation bits (110 bits) for a total of 128 bits
where the combination and significand bits are calculated based on these templates:
![image](https://user-images.githubusercontent.com/44870738/56422006-fe7d4e80-62ad-11e9-8655-a3dfcd6c5ce6.png)

Reply:
wiki: https://en.wikipedia.org/wiki/Decimal128_floating-point_format

Reply:
So, based on the test, the big value in `second form` are just converted to `0`. Is it expected? @jyemin, @rstam. Can you take a look at here?

Reply:
I noticed that the numbers in the `first form` with exponent < '-56' also always converted to '0'. For the income value in the above test, the `exponent` is equal to '-6176'. 
So, it looks like we set '0' in case if the number is out of the `decimal` bounds?

Reply:
I'm not sure about this.  Is it necessary to understand in order to proceed with the change you made?  It seems that this is testing an unchanged code path.

Reply:
yep, I didn't change this code path. I noticed that we don't have tests for this functionality(for `"second"` number form) when I investigated how conversion logic works. I thought maybe the same logic could be added in several places. Currently, I think that it's enough to have only changes which I've already added, but at the same time, I think adding tests which were skipped before, it's a good idea. I agree that we don't lose a precision here (as described in the ticket title), but we lose the whole number, that in some meaning seems similar to me.

So, as a summary for this comment, I left this question more to have an additional set of eyes at here than a real question.